v2/tools/generator/internal/config/object_model_configuration.go

```diff
@@ -29,14 +29,92 @@ import (
 type ObjectModelConfiguration struct {
 	groups      map[string]*GroupConfiguration // nested configuration for individual groups
 	typoAdvisor *typo.Advisor
+
+	// Group access fields here (alphabetical, please)
+	PayloadType groupAccess[PayloadType]
+
+	// Type access fields here (alphabetical, please)
+	AzureGeneratedSecrets typeAccess[[]string]
+	DefaultAzureName      typeAccess[bool]
+	Export                typeAccess[bool]
+	ExportAs              typeAccess[string]
+	GeneratedConfigs      typeAccess[map[string]string]
+	Importable            typeAccess[bool]
+	IsResource            typeAccess[bool]
+	ManualConfigs         typeAccess[[]string]
+
+	ResourceEmbeddedInParent typeAccess[string]
+	SupportedFrom            typeAccess[string]
+	TypeNameInNextVersion    typeAccess[string]
+
+	// Property access fields here (alphabetical, please)
+	ARMReference                   propertyAccess[bool]
+	ImportConfigMapMode            propertyAccess[ImportConfigMapMode]
+	IsSecret                       propertyAccess[bool]
+	ResourceLifecycleOwnedByParent propertyAccess[string]
+}
+
+type groupAccess[T any] struct {
```

Comment:
Go supports multiple type parameters, right?

Could we have a single type: `type accessor[TConfiguration any, TProperty any]`? I think the issue here would be that there are slightly different Lookup implementations for each of the group/type/property variants, so I guess this isn't possible unless you also had `accessor` take a lambda that did the lookup (making it more awkward to construct), or possibly made `TConfiguration` require to implement an interface where that interface impl had some of the lookup logic?

Obviously pretty minor but just when I was looking at these types there' basically identical in shape here except for the first parameter of `accessor`

Reply:
Different Lookup() signatures too, so hard to abstract. I did try!